portfolio/src/main/webapp/script.js

```diff
@@ -119,3 +138,46 @@ function setMoodTo(mood) {  // eslint-disable-line no-unused-vars
 
   document.getElementById(mood).style.display = 'block';
 }
+
+function createMap() {
+  map = new google.maps.Map(
+      document.getElementById('map'), {center: {lat: 20, lng: 0}, zoom: 1});
+
+  google.maps.event.addListener(map, 'click', function(event) {
+    postMarker(event.latLng);
+  });
+
+  getMarkers();
+}
+
+function getMarkers() {
+  fetch('/marker').then((response) => response.json()).then((markerJsons) => {
+    markerJsons.forEach((marker) => {
+      placeMarker(JSON.parse(marker));
+    });
+  });
+}
+
+function postMarker(location) {
+  placeMarker(location);
+  const request =
+      new Request('/marker?json=' + JSON.stringify(location), {method: 'POST'});
+  fetch(request);
+  markersCreated++;
+}
+
+function placeMarker(location) {
+  new google.maps.Marker({position: location, map: map});
+}
+
+/**
+ * Deletes the most recent marker placed. Only markers placed since the last
+ * page refresh can be deleted.
+ */
+function deleteRecentMarker() {  // eslint-disable-line no-unused-vars
+  if (markersCreated > 0) {
```

Comment:
This is a nice little feature on your site. But do be aware that we would not bother doing a check like this as visitors can just change the value of markersCreated in their browser console as they wish, which makes the check meaningless. No need to update the code here; I want you to be aware of this.

Reply:
I didn't realize that, interesting!

Reply:
Yep - I went on your site and deleted all the previous markers 😄

Reply:
Aww man now I have to put those back on